app/controllers/deactivates_controller.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class DeactivatesController < ApplicationController
+  def destroy
+    @user = authorize User.find_by(id: params[:user_id]), :edit?
```

Comment:
Szerintem itt nem a megfelelő policy-t, és annak sem a megfelelő segédfüggvényét használod.
```ruby
authorize User.first, :edit?           # UserPolicy....edit?
# Nekünk a DeactivatePolicy....destroy? kellene
```

Reply:
Légyszi ellenőrizd le, hogy tényleg abba a policy segédfüggvénybe érkezel-e, amikor ez a sor fut, mint ahova szeretnél. pl.: Ha raksz egy bybug-ot a DeactivatePolicy#destroy?-ba, akkor megállunk-e ott?

Szerintem a mostani kód nem oda fog mutatni.

Az `authorize` függvény defniníciója (ctrl + click):
```ruby
def authorize(record, query = nil, policy_class: nil)
  # stuff
end
```
Mindenképp vár egy `record` -ot, amiből általában ő ki is tudja neked találni, hogy melyik PolicyClass-t szeretnéd használni majd authorizációra. Viszont meg is adja neked a lehetőséget, hogy ezt a PolicyClass-t felülbíráld.
Mi ugye User record-ot küldünk be, ami alapján ő a UserPolicy-t fogja használni. Viszont mi azt szeretnénk, hogy a DeactivatePolicy-t használja.

`authorize` hívásakor az a default működés, hogy amelyik kontroller akcióból hívtad, a megfelelő PolicyClass korreláló segédfüggvényét fogja meghívni. Magyarul ha a `show` akcióból hívom, akkor a `show?` segédfüggvényt fogja használni. Ezt nem kell külön megadni. Csak akkor kell explicit kiírni, hogy melyik segédfüggvényt szeretnéd használni, ha az eltér a kontroller akciótól. pl.:
```ruby
def show
  # UserPolicy#show?
  # Ez a default működés, nincs szükség arra, hogy kiírjuk a :show? -t
  authorize User.find_by(id: params[:id])
end

def action_with_custom_name
  # UserPolicy#action_with_custom_name?
  # Nem csak a standard neveket ismeri. Itt is megtalálja neked a megfelelő függvényt, ha definiálva van az :action_with_custom_name? a UserPolicy-ban
  authorize User.find_by(id: params[:id])
end

def some_complex_action
  # UserPolicy#show?
  # Egy komplex akcióban le akarom ellenőrizni, hogy látható-e a felhasználó számára az adott record. Nem akarom figyelembe venni a kontroller akciót, mert az egy komplex működésért felelős, nem feleltethető meg egy az egyben a policy-nak, amit ellenőrizni szeretnék.
  authorize User.find_by(id: params[:id]), :show?
end
```